types/inquirer/index.d.ts

```diff
@@ -288,7 +358,7 @@ declare namespace inquirer {
             extra?: any;
             key?: string;
             checked?: boolean;
-            disabled?: string | (<T = Answers>(answers: T) => any);
+            disabled?: string | (<A>(answers: A) => any);
```

Comment:
Please consider to add a generic parameter to `ChoiceOption`, `Choice`, `Choices` and `ChoiceType` rather than adding a generic parameter to a callbacks and constructors.
This would make it ways better compatible with `QuestionOptions<A>` and more comfortable to use, I think.

Reply:
A bit out of scope, but I'll do it.